src/Functions/FunctionFile.cpp

```diff
@@ -0,0 +1,125 @@
+#include <Columns/ColumnString.h>
+#include <Columns/IColumn.h>
+#include <Functions/FunctionFactory.h>
+#include <DataTypes/DataTypeString.h>
+#include <IO/ReadBufferFromFile.h>
+#include <Poco/File.h>
+#include <Poco/Path.h>
+#include <Interpreters/Context.h>
+#include <unistd.h>
+
+namespace DB
+{
+
+    namespace ErrorCodes
+    {
+        extern const int ILLEGAL_COLUMN;
+        extern const int NOT_IMPLEMENTED;
+        extern const int FILE_DOESNT_EXIST;
+        extern const int CANNOT_OPEN_FILE;
+        extern const int CANNOT_CLOSE_FILE;
+        extern const int CANNOT_READ_FROM_FILE_DESCRIPTOR;
+        extern const int INCORRECT_FILE_NAME;
+        extern const int DATABASE_ACCESS_DENIED;
+    }
+
+    /** A function to read file as a string.
+  */
+    class FunctionFile : public IFunction
+    {
+    public:
+        static constexpr auto name = "file";
+        static FunctionPtr create(const Context &context) { return std::make_shared<FunctionFile>(context); }
+        explicit FunctionFile(const Context &context_) : context(context_) {}
+
+        String getName() const override { return name; }
+
+        size_t getNumberOfArguments() const override { return 1; }
+        bool isInjective(const ColumnsWithTypeAndName &) const override { return true; }
+
+        DataTypePtr getReturnTypeImpl(const ColumnsWithTypeAndName & arguments) const override
+        {
+            if (!isString(arguments[0].type))
+                throw Exception(getName() + " is only implemented for types String", ErrorCodes::NOT_IMPLEMENTED);
+            return std::make_shared<DataTypeString>();
+        }
+
+        bool useDefaultImplementationForConstants() const override { return true; }
+        ColumnNumbers getArgumentsThatAreAlwaysConstant() const override { return {1}; }
```

Comment:
I see that non-constant argument is supported (as expected). So, this is line is unneeded.

Reply:
Remove it soon.